packages/api/src/session.ts

```diff
@@ -20,10 +20,26 @@ const DELETE_SESSION = 'com.atproto.session.delete'
 const CREATE_ACCOUNT = 'com.atproto.account.create'
 
 export class SessionClient extends Client {
+  private cache: Map<string, SessionServiceClient> = new Map()
+
   service(serviceUri: string | URL): SessionServiceClient {
+    const cacheKey = toCacheKey(serviceUri)
+    const inst = this.cache.get(cacheKey)
+    if (inst) {
+      return inst
+    }
     const xrpcService = new SessionXrpcServiceClient(this.xrpc, serviceUri)
-    return new SessionServiceClient(this, xrpcService)
+    const service = new SessionServiceClient(this, xrpcService)
+    this.cache.set(cacheKey, service)
+    return service
+  }
+}
+
+function toCacheKey(uri: string | URL) {
+  if (typeof uri === 'string') {
+    uri = new URL(uri)
   }
+  return uri.hostname
```

Comment:
It's unlikely to come-up, but `uri.host` might be marginally better since it takes into account the port.  An example might be different services running on `localhost`, e.g. in tests.

Reply:
If you want:
```suggestion
  return uri.host
```